lib/effects/effects.dart

```diff
@@ -29,9 +29,14 @@ abstract class ComponentEffect<T extends Component> {
   final bool isRelative;
   final bool _initialIsInfinite;
   final bool _initialIsAlternating;
-  double percentage;
-  double curveProgress;
-  double peakTime;
+  double? percentage;
+
+  /// FIXME: is it a safe default?
+  double curveProgress = 0.0;
+
+  /// FIXME: Where it's been set? Let's try defaulting it to 0.0 for now as at least
+  /// some subclasses set it in `initialize` method
+  double peakTime = 0.0;
```

Comment:
@luanpotter does it look ok to you to use a default value here?

Reply:
It should be fine here too since it is initialized before the first `update` is called.